src/lib/src/select-container.component.ts

```diff
@@ -239,18 +246,45 @@ export class SelectContainerComponent implements AfterViewInit, OnDestroy {
   }
 
   private observeSelectableItems() {
-    // Update the container as well as all selectable items if the list has changed
-    this.$selectableItems.changes.pipe(takeUntil(this.destroy$)).subscribe((items: QueryList<SelectItemDirective>) => {
-      const newList = items.toArray();
-      const removedItems = this._selectedItems.filter(item => !newList.includes(item.value));
+    // Listen for update actions and either select or deselect an item
+    this.updateItems$
+      .pipe(
+        mergeMap(update => {
+          const item = update.item;
+
+          switch (update.type) {
+            case UpdateActions.Add:
+              return this.addItem(item).pipe(tap(itemAdded => itemAdded && item.select()));
+            case UpdateActions.Remove:
+              return this.removeItem(item).pipe(tap(itemRemoved => itemRemoved && item.deselect()));
+          }
+        }),
+        takeUntil(this.destroy$)
+      )
+      .subscribe();
 
-      setTimeout(() => {
-        if (removedItems.length) {
-          removedItems.forEach(item => this.removeItem(item));
-        }
-        this.update();
-      });
-    });
+    // Update the container as well as all selectable items if the list has changed
+    this.$selectableItems.changes
+      .pipe(
+        withLatestFrom(this._selectedItems$),
+        map(([items, selectedItems]: [QueryList<SelectItemDirective>, any[]]) => {
+          const newList = items.toArray();
+          const removedItems = selectedItems.filter(item => !newList.includes(item.value));
+
+          return removedItems;
+        }),
+        concatMap(removedItems => {
+          if (removedItems.length) {
+            return forkJoin(removedItems.map(item => this.removeItem(item)));
+          }
+
+          return empty();
+        }),
+        tap(() => this.update()),
+        takeUntil(this.destroy$),
+        observeOn(async)
```

Comment:
For the function in the `tap` to be called on the `async` scheduler, the `observeOn` will need to precede the `tap`.